src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -1233,12 +1427,189 @@ def test_handling_nested_intrinsic():
     assert not cblocks
 
 
-@pytest.mark.xfail(reason="#412 Fortran array notation not yet handled in "
-                   "non-NEMO PSyIR")
+@pytest.mark.usefixtures("f2008_parser")
+def test_array_section():
+    ''' Check that we correctly handle an array section '''
+
+    def _array_create(code):
+        '''Utility function that takes the supplied executable code and
+        returns its PSyIR representation.
+
+        :param str code: the executable code as a string.
+
+        :returns: the executable code as PSyIR nodes.
+        :rtype: :py:class:`psyclone.psyir.nodes.node`
+
+        '''
+        processor = Fparser2Reader()
+        fake_parent = Node()
+        reader = FortranStringReader(code)
+        fp2node = Execution_Part.match(reader)[0][0]
+        processor.process_nodes(fake_parent, [fp2node])
+        return fake_parent.children[0].children[0]
+
+    def _check_array(node, ndims):
+        '''Utility function that checks that the supplied node is an array and
+        has the expected number of dimensions. Also checks that the
+        node is the parent of its children.
+
+        :param node: the node to check.
+        :type node: :py:class:`psyclone.psyir.nodes.array`
+        :param int ndims: the number of expected array dimensions.
+
+        '''
+        assert isinstance(node, Array)
+        assert len(node.children) == ndims
+        for child in node.children:
+            assert child.parent is node
+
+    def _check_range(array, dim):
+        '''Utility function that checks that the "dim" index of the supplied
```

Comment:
You're assuming 'dim' is indexed from 1 here - please say so explicitly (and in the routine above too).

Reply:
Added a comment in both routines where `dim - 1` is used.